EIPS/eip-7805.md

```diff
@@ -99,17 +98,25 @@ For each transaction `T` in ILs, perform the following:
 
   - If `T` is invalid, then continue to the next transaction. 
   
-  - If `T` is valid, terminate process and assert block `B` as invalid.
+  - If `T` is valid, terminate process and return an error.
 
 3. Execute `T` on state `S`. Assert that the execution of `T` fails.
 
-If `B` is full, the process terminates. Also note that we do not need to reset the state to `S`, since the only way for a transaction to alter the state is for it to execute successfully, in which case the block is invalid, and so the block will not be applied to the state.
+If `B` is full, the process terminates. Also note that the state `S` must be restored if a transaction from the IL is successfully executed against it.
+
+#### Engine API Changes
 
 We make the following changes to the engine API:
 
-- Add `engine_getInclusionList` endpoint to retrieve an IL from the `ExecutionEngine`
-- Modify `engine_newPayload` endpoint to include a parameter for transactions in ILs determined by the proposer
-- Modify `engine_forkchoiceUpdated` endpoint to include a field in the payload attributes for transactions in ILs determined by the proposer
+- Add `engine_getInclusionListV1` endpoint to retrieve an IL from the `ExecutionEngine`.
+- Add `engine_updatePayloadWithInclusionListV1` endpoint to update a payload with the IL that should be used to build the block. This takes as an argument an 8-byte `payloadId` of the ongoing payload build process, along with the IL itself.
+- Modify `engine_newPayload` endpoint to include a parameter for transactions in ILs determined by the proposer. If the IL is not satisfied an `INVALID_INCLUSION_LIST` error must be returned.
```

Comment:
```suggestion
- Modify `engine_newPayload` endpoint to include a parameter for transactions in ILs determined by the validator. If the IL is not satisfied an `INVALID_INCLUSION_LIST` error must be returned.
```

or "attester"? it is not the proposer that supplies the IL that should be satisfied by the proposer's payload.

Reply:
Or we could use "the IL committee members"